driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -111,7 +111,7 @@ private void startNewExecution() {
             return;
 
         Message.Request request = callback.request();
-        int position = executionCount.incrementAndGet();
+        int position = executionIndex.getAndIncrement();
```

Comment:
Changed this because it's more consistent with the way things are exposed in ExecutionInfo.
The initial execution will now have a position of 0 instead of 1. This was only exposed in debug logs so I don't think it's an issue.